src/NuGet.Core/NuGet.Protocol/Resources/ServiceIndexResourceV3.cs

```diff
@@ -160,6 +164,14 @@ private static IDictionary<string, List<ServiceIndexEntry>> MakeLookup(JObject i
                         continue;
                     }
 
+                    // Capture if the resource is http & the source is https
+                    var resourceIsHttp = uri.Scheme == Uri.UriSchemeHttp && uri.Scheme != Uri.UriSchemeHttps;
```

Comment:
Isn't `Uri.UriSchemeHttp` enough? These are strings not enums, so the value can't be both UriSchemeHttp and UriSchemeHttps, anyway.